index.d.ts

```diff
@@ -0,0 +1,33 @@
+interface ClearablePromise<T> extends Promise<T> {
+	/**
+	 * Clears the delay and settles the promise.
+	 *
+	 * @memberof ClearablePromise
+	 */
+	clear(): void;
+}
+
+declare const delay: {
+	/**
+	 * Create a promise which resolves after the specified `milliseconds`.
+	 * Optionally pass a `value` to resolve.
+	 *
+	 * @param {number} milliseconds Milliseconds to delay the promise.
+	 * @param {T} [value] Value to resolve in the returned promise.
+	 * @returns {ClearablePromise<T>} a promise which resolves after the specified `milliseconds`
+	 */
+	<T = never>(milliseconds: number, value?: T): ClearablePromise<T>;
+
+	/**
+	 * Create a promise which rejects after the specified `milliseconds`.
+	 * Optionally pass a `reason` to reject.
+	 *
+	 * @param {number} milliseconds Milliseconds to delay the promise.
```

Comment:
Is it required to provide the types here? I feel it's moot as it's already defined in the signature and TS could just get it from there. I don't like needless boilerplate.

Reply:
Seems it's not: https://github.com/Microsoft/tsdoc

Reply:
According to TSDoc docs, there should be `-` between the param name and description.

Reply:
Yes, should be removed.